Tidy the install process and RPM spec

jthiltges opened the pull request:
- Reduce the dependencies for the unauthenticated xrootd
- Create the config and systemd unit files during the `make install` process
- Tidy the stashcache configuration
    - Reduce the default disk usage of the cache to 90/95% to avoid accidentally filling filesystems too full
    - Move the certificate configuration inside the auth instance of xrootd

Hi @zvada. I'd appreciate your feedback when you have a moment.

zvada commented:
Nice! Have you tried build against new changes and rpm install, @jthiltges?

bbockelm commented on spec/stashcache.spec:
```diff
@@ -84,12 +110,19 @@ rm -rf %{_buildroot}
 %files origin-server
-%config(noreplace) %{_sysconfdir}/xrootd/xrootd-stashcache-origin-server.cfg
-%config(noreplace) %{_sysconfdir}/xrootd/xrootd-stashcache-origin-server-itb.cfg
+%attr(-, xrootd, xrootd) %config(noreplace) %{_sysconfdir}/xrootd/xrootd-stashcache-origin-server.cfg
```
Is this right?  I would prefer if xrootd didn't have the ability to edit its own configuration file.

jthiltges replied:
I'd followed the upstream pattern of making the config files owned by xrootd:
https://github.com/xrootd/xrootd/blob/master/packaging/rhel/xrootd.spec.in#L711

I agree it does seem unsettling in the new write-enabled xrootd world. Is that worth opening a PR with the xrootd folks to change?

bbockelm replied:
Erm - I think I'd just prefer to diverge from upstream practice here.

jthiltges commented:
I think the xrootd-stashcache-cache-server.cfg is now caught up with the improvements from Mátyás. Here's a summary of the differences:

- Set `pfc.ram 7g` to allow a bit of room for the OS on 8GB system (documented minimum RAM)
- Added `pss.origin redirector-itb` as a commented line, rather than adding a separate itb config file
- Set `pfc.diskusage 0.90 0.95` to avoid filling the cache filesystem too far, because ZFS in particular gets unhappy.
  Perhaps the default should be lower yet, as ZFS beyond 80-85% may start to see performance issues.
- Moved the `http.*` certificate config, `xrootd.seclib`, and `sec.protocol` inside the auth config block

matyasselmeci commented on spec/stashcache.spec:
```diff
@@ -9,18 +9,18 @@ BuildArch: noarch
 Source0:   %{name}-%{version}.tar.gz
 
 BuildRoot: %(mktemp -ud %{_tmppath}/%{name}-%{version}-%{release}-XXXXXX)
```
As long as we're in here, we should remove this line because it was only required for RHEL 5.

matyasselmeci commented on spec/stashcache.spec:
```diff
@@ -9,18 +9,18 @@ BuildArch: noarch
+BuildRequires: systemd
+%{?systemd_requires}
 
 %if ! (0%{?fedora} > 12 || 0%{?rhel} > 5)
```
Same with this block.

matyasselmeci commented on spec/stashcache.spec:
```diff
@@ -65,14 +96,9 @@ exit 1
+# Create xrootd certificate directory
+mkdir -p %{buildroot}%{_sysconfdir}/grid-security/xrd
 
 %clean
```
%clean is also only for RHEL5.

matyasselmeci replied:
as is %defattr(-,root,root) below but GitHub won't let me put a line comment there.

jthiltges commented on spec/stashcache.spec:
```diff
@@ -8,15 +8,6 @@ URL:       https://opensciencegrid.github.io/StashCache/
 Source0:   %{name}-%{version}.tar.gz
 
-BuildRoot: %(mktemp -ud %{_tmppath}/%{name}-%{version}-%{release}-XXXXXX)
-BuildRequires: systemd
```
I wasn't certain, and it did still build OK without it.
I'll put it back. Thanks! :-)

matyasselmeci commented:
@zvada, @jthiltges, is it OK if I resolve the conflicts and merge this?

jthiltges commented:
Yes, please, for my vote

matyasselmeci commented:
https://opensciencegrid.atlassian.net/browse/SOFTWARE-3434

zvada commented:
What I've been following so far - yes from me - merge, please! Nice work!
